### ISSUE TITLE
Label is not announced if the input has a value (VoiceOver & Safari)

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-text-field#324]
### Description
When focusing a text field, which has a label and a value, using macOS VoiceOver in Safari only the value is announced but not the label. Both are announced when using VoiceOver and Chrome.

When the field only has a label but no value, the label is announced.

### Expected outcome
Both the label and value should be announced. 

For example, if the component is configured as:
```html
<vaadin-text-field label="Name" value="John"></vaadin-text-field>
```
VoiceOver should announce:
```
John Insertion at end of text. Name, edit text
(pause)
You are currently on a text field, inside of web content. To enter text in this field ...
```

### Actual outcome
VoiceOver announces:
```
John 
(pause)
You are currently on a text field, inside of web content. To enter text in this field ...
```

### Live Demo
https://cdn.vaadin.com/vaadin-text-field/2.3.2/demo/#text-field-basic-demos

### Steps to reproduce
1. Turn on VoiceOver <kbd>Cmd</kbd>+<kbd>F5</kbd>
1. Use <kbd>Tab</kbd> to focus a field with a label and a value

### Browsers Affected

- [ ] Chrome
- [ ] Firefox
- [x] Safari
- [ ] Edge
- [ ] IE 11
- [x] iOS Safari
- [ ] Android Chrome

**Comment by anezthes** [comment]
Wasn't able reproduce this 100% of the time. I don't know the inner workings on the compoennt but the outmost `tabindex="0"` seems unnecessary. Removing it also seems to fix this issue.
